### PR TITLE
Adds standard user driver delegate method for responding to Version History button

### PR DESCRIPTION
**Pull request by billymeltdown**
Adds a new delegate method to the `SPUStandardUserDriver` protocol for responding to the Version History button with custom behavior. Standard user driver first checks for the presence of the delegate method, then falls back to opening the release notes link as the button action as before.

As discussed in #1877 

## Checklist:

- [x] My change is being tested and reviewed against the Sparkle 2.x branch. New changes must be developed on the 2.x development branch first.
- [ ] My change is being backported to master branch (Sparkle 1.x). Please create a separate pull request for 1.x, should it be backported. Note 1.x is feature frozen and is only taking bug fixes, localization updates, and critical OS adoption enhancements.
- [x] I have reviewed and commented my code, particularly in hard-to-understand areas.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] My change is or requires a documentation or localization update

## Testing

I tested and verified my change by using one or multiple of these methods:

- [ ] Sparkle Test App
- [ ] Unit Tests
- [x] My own app
- [ ] Other (please specify)

I did a `make release` of my fork with my changes, dropped that `Sparkle.framework` into my app in place of current version, and checked that delegate method is called correctly. Then removed delegate method and confirmed previous behavior worked as expected.

macOS version tested: 11.6

### CODE REVIEW

File: Sparkle/SPUStandardUserDriver.m [hunk 1]
```diff
@@ -274,9 +274,22 @@ - (void)showUpdateNotFoundWithError:(NSError *)error acknowledgement:(void (^)(v
...
+
+                // Check if the delegate implements a Version History action
+                id <SPUStandardUserDriverDelegate> delegate = self.delegate;
+                
+                if ([delegate respondsToSelector:@selector(standardUserDriverShowVersionHistory)]) {
+                    [alert addButtonWithTitle:localizedButtonTitle];
+                    
+                    __weak SPUStandardUserDriver *weakSelf = self;
+                    secondaryAction = ^{
+                        [weakSelf.delegate standardUserDriverShowVersionHistory];
```
[the remark this answers is not in the document]
Comment (billymeltdown) [reply Sparkle/SPUStandardUserDriver.m]: Ah, had a feeling I was overdoing it! Removed.

File: Sparkle/SPUStandardUserDriverDelegate.h [hunk 2]
```diff
@@ -45,4 +45,11 @@ SU_EXPORT @protocol SPUStandardUserDriverDelegate <NSObject>
  */
 - (_Nullable id <SUVersionDisplay>)standardUserDriverRequestsVersionDisplayer;
 
+/**
+ Called if implemented when the Version History button is chosen by the user. Otherwise,
+ the `releaseNotesFullLink` or `releaseNoteLink` in the appcast's latest item will be be
+ opened.
+ */
+- (void)standardUserDriverShowVersionHistory;
```
[the remark this answers is not in the document]
Comment (billymeltdown) [reply Sparkle/SPUStandardUserDriverDelegate.h]: That's a nice idea, added that in too, still need to verify in my app.

**Comment by billymeltdown** [comment]
I've just re-tested these edits in my own app to verify the changes.

**Comment by zorgiepoo** [comment]
Thank you. I may do some minor commenting tweaks, but looks good.

**Comment by aonez** [comment]
That was quick!

**Comment by billymeltdown** [comment]
@aonez Happy to help, even if it's just a little bit 😄

Modify away @zorgiepoo! I tried to ape the comments style from the rest of the project but figured they would need work.

**Comment by zorgiepoo** [comment]
Any little help is good :).

@aonez let me know if you're able to adjust SUAppcastItem.m / SPUStandardUserDriver.m for the `fullReleaseNotesLink` change, otherwise I may be able to get to it over the weekend. I've been thinking I think it's still good to provide that option in the feed.

**Comment by aonez** [comment]
Sure I had other things but I’ll try to do it today

**Comment by aonez** [comment]
@zorgiepoo I've made a [pull](https://github.com/sparkle-project/Sparkle/pull/2001). It lacks `generate_appcast` code, let's follow there.

**Pull request merged**